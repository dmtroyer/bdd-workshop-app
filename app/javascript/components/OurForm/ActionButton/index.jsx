import React from 'react';
import { Button } from 'reactstrap';
import PropTypes from 'prop-types';

const ActionButton = ({ children, isSubmitting }) => (
  <Button name="submit" color="primary" type="submit" disabled={isSubmitting} className="float-right">
    {children}
  </Button>
);

ActionButton.propTypes = {
  children: PropTypes.node,
  isSubmitting: PropTypes.bool,
};

export default ActionButton;
