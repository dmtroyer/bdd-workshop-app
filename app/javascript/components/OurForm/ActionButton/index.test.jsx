import React from 'react';
import { shallow } from 'enzyme';
import { Button } from 'reactstrap';
import ActionButton from '.';

describe('ActionButton', () => {
  it('Renders a primary submit button with children text', () => {
    const wrapper = shallow(<ActionButton>button text</ActionButton>);
    expect(wrapper).toMatchSnapshot();
    expect(
      wrapper
        .find(Button)
        .dive()
        .text()
    ).toEqual('button text');
  });

  it('Renders button NOT disabled by default', () => {
    const wrapper = shallow(<ActionButton>button text</ActionButton>);
    expect(wrapper.find(Button).prop('disabled')).toBeFalsy();
  });

  it('Renders button disabled if isSubmitting', () => {
    const wrapper = shallow(
      <ActionButton isSubmitting={true}>button text</ActionButton>
    );
    expect(wrapper.find(Button).prop('disabled')).toBeTruthy();
  });
});
