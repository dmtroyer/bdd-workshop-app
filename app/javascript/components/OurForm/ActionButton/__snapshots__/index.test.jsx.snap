// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton Renders a primary submit button with children text 1`] = `
<Button
  color="primary"
  name="submit"
  tag="button"
  type="submit"
>
  button text
</Button>
`;
